template/src/context/AuthContext.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import cache from '../utils/cache';
+
+type AuthContextData = {
+  isAuthenticated: boolean;
+  authenticate: ({
+    token,
+    persist,
+  }: {
+    token: string;
+    persist?: boolean;
+  }) => void;
+  logout: () => void;
+};
+
+type ProviderProps = {
+  children: React.ReactNode;
+};
+
+export const AuthContext = React.createContext<AuthContextData>(
+  {} as AuthContextData
+);
+
+export function AuthProvider({ children }: ProviderProps) {
+  const { setCache } = cache;
+  const [isAuthenticated, setIsAuthenticated] = React.useState(false);
+
+  function authenticate({
+    token,
+    persist,
+  }: {
+    token: string;
+    persist?: boolean;
```

Comment:
I don't see any use for this `persist` property/variable in this login case. This action assumes that the token will always be stored in `localStorage`, so the `persist` value would always be true. Am I missing something?